### ISSUE TITLE
Alphanumeric Sender ID in Laravel 4.2

### ISSUE THREAD

**Issue by dror3go** [aloha/laravel-twilio#75]
Hi,

I'm currently using v1.0.2. Is it possible to send text messages so that the recipient will see [a textual "from"](https://www.twilio.com/help/faq/sms/what-is-alphanumeric-sender-id-and-how-do-i-get-started) instead of my purchased Twilio number?
If this feature isn't supported in v1.0.2 - which stable version available for Laravel 4.2 should I use?

Thanks

**Comment by hannesvdvreken** [comment]
You can use any version of Laravel 4.2, I suggest using the latest. It has no effect on your wish to use a textual from field. I guess, but not 100% sure, that you can use twilio settings to use a textual from field.

**Comment by dror3go** [comment]
According to Twilio's support, it's not something to set in the account's settings, but simply a change in the code.

When I'm changing the "from" in *config/packages/aloha/twilio/twilio.php* to a textual string instead of my Twilio number, I'm getting an error:
> The From phone number 869669 is not a valid, SMS-capable inbound phone number or short code for your account.

Where "869669" is the **message** I'm trying to send, not the "from".
Note that when changing back the config value of "from" to my Twilio number - the SMS is sent.

BTW, this feature is pretty new in Twilio - they introduced it in May 2015.

**Comment by hannesvdvreken** [comment]
Which minimum SDK version is required for that? If it's version 4.x.y you will need to install [2.0.0-RC3](https://github.com/aloha/laravel-twilio/releases/tag/2.0.0-RC3) minimum to be able to use this. No worries, it works with Laravel 4.2 as well.

**Comment by hannesvdvreken** [comment]
Please add some links to the twilio documentation, if possible.

**Comment by dror3go** [comment]
I haven't noticed any information about minimun SDK version. The links I got from their support are these for the FAQ:
https://www.twilio.com/help/faq/sms/what-is-alphanumeric-sender-id-and-how-do-i-get-started
https://www.twilio.com/help/faq/sms/can-i-specify-the-phone-number-a-recipient-sees-when-getting-an-sms-from-my-twilio-app
https://www.twilio.com/help/faq/sms/what-is-sticky-sender-and-how-does-it-work

This is the documentation detailing about this: https://www.twilio.com/docs/api/rest/sending-messages

**Comment by hannesvdvreken** [comment]
I don't have the time to help you, maybe you can open an issue on the [Twilio SDK repo](https://github.com/twilio/twilio-php/)?

Please report back if you found the solution.

**Comment by dror3go** [comment]
After doing some test, I came to a conclusion that this is an issue in this package, not in Twilio SDK: When I send an SMS directly with `new \Services_Twilio()` - I'm able to send the message with alphanumeric sender ID. When I use this package - I'm getting an error `SMS-capable inbound phone number or short code for your account`.

@hannesvdvreken can you please reopen this issue?

**Comment by hannesvdvreken** [comment]
Please report back on how you used the `\Services_Twilio` and how it differs from how this package calls it.

**Comment by dror3go** [comment]
I used the exact example as appears in [Twilio's documentation](https://twilio-php.readthedocs.org/en/latest/):
```
// No need to require Services/Twilio.php
$client = new \Services_Twilio($sid, $token);
$message = $client->account->messages->sendMessage(
  'MY_ALPHANUMERIC_SENDER_ID', // From a valid Twilio number
  '<MY_PHONE_NUMBER>', // Text this number
  "Hello monkey!"
);
```

**Comment by hannesvdvreken** [comment]
I suggest you add some logging here and there in the [Twilio class](https://github.com/aloha/laravel-twilio/blob/master/src/Twilio.php) to see if all Twilio classes are called with the correct parameters.

**Comment by dror3go** [comment]
Mystery solved: the version used was 1.0.2, which is way outdated. Using current version fixed the problem. Thanks again.

**Comment by hannesvdvreken** [comment]
Good find.

The SDK version you linked [documentation](https://twilio-php.readthedocs.org/en/latest/) from is 4.2.1.

> This documentation is for version 4.2.1 of twilio-php.

In order to use this wrapper package with Twilio SDK 4.2.1 you need to install 2.0.0 at the least (`^2.0`).
